### ISSUE TITLE
Simple questions about robustness testing and generated image quality

### ISSUE THREAD

**Issue by Jayamoon59** [facebookresearch/stable_signature#4]
Hi, I am replicating your code, your work is really comprehensive and wonderful, it is very helpful to me, but I would like to ask some small questions.

I've done the basics of generating a watermarked image (Figure 6 of the article), but why is it not as high quality as the image in the article, is it related to the data set? I would also like to know if there is code to extract embedded fixed watermarks?

The other problem is that I want to implement robust testing, which part of this is different from the most basic build setup? Or is the robustness tested after the image is generated?

Hope to get your reply, thank you for your time.

**Comment by pierrefdz** [comment]
Thanks!

> I've done the basics of generating a watermarked image (Figure 6 of the article), but why is it not as high quality as the image in the article, is it related to the data set? I would also like to know if there is code to extract embedded fixed watermarks? 

Could you give a link to your images and setup used for fine-tuning the LDM decoder?
You can try playing a bit with the hyper-parameters of the LDM decoder fine-tuning (length of training, weight of the perceptual loss), depending on the dataset this could be different.

> The other problem is that I want to implement robust testing, which part of this is different from the most basic build setup? Or is the robustness tested after the image is generated?

The code necessary for running the evaluations of the paper is available in the `run_evals.py` file. I've documented it in the readme: https://github.com/facebookresearch/stable_signature/tree/main#decode-and-evaluate.
By default this run image quality and watermark robustness metrics on many augmentations. You can change the augmentations and remove the image evaluation if you don't need it.

You can use the argument `--decode_only True` in this script that allows to retrieve messages of all images in a folder, without computing the bit accuracies.

**Comment by pierrefdz** [comment]
Closing because of inactivity, feel free to re-open
